wman-backend/Wman.Logic/Classes/EventLogic.cs

```
@@ -351,6 +351,57 @@ public async Task DnDEvent(int Id, DnDEventDTO newWorkEvent)
                 throw new InvalidOperationException(WmanError.EventDateInvalid);
             }
         }
+        public async Task<WorkEventForWorkCardDTO> StatusUpdater(int eventId, string username)
+        {
+            var workevent = await eventRepo.GetOneWithTracking(eventId);
+
+            if (Status.finished == workevent.Status)
+            {
+                throw new InvalidOperationException(WmanError.StatusFinished);
+            }
+            if (Status.proofawait == workevent.Status && workevent.ProofOfWorkPic.Count == 0)
+            {
+                throw new InvalidOperationException(WmanError.StatusPowMissing);
+            }
+            switch (workevent.Status)
+            {
+                case Status.awaiting:
+                    workevent.WorkStartDate = DateTime.Now;
+                    break;
+                case Status.started:
+                    break;
+                case Status.proofawait:
+                    workevent.WorkFinishDate = DateTime.Now;
+                    break;
+                case Status.finished:
+                    break;
+                default:
+                    break;
+            }
```

Do not use both `if-else` and `switch-case` though you can use an `if` in a `case`. If you choose `switch-case` the default case should throw an `Exception` like `Status not found`.